unicorn/include/cocaine/unicorn/api/zookeeper.hpp

```diff
@@ -0,0 +1,173 @@
+/*
+* 2015+ Copyright (c) Anton Matveenko <antmat@yandex-team.ru>
+* All rights reserved.
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*/
+
+#ifndef COCAINE_UNICORN_ZOOKEEPER_API_HPP
+#define COCAINE_UNICORN_ZOOKEEPER_API_HPP
+
+#include <cocaine/idl/unicorn.hpp>
+#include <cocaine/zookeeper/connection.hpp>
+#include <cocaine/unicorn/writable.hpp>
+#include "cocaine/unicorn/api.hpp"
+
+namespace cocaine { namespace unicorn {
+
+zookeeper::cfg_t make_zk_config(const dynamic_t& args);
+
+class zookeeper_api_t : public api_t {
+public:
+
+    struct context_t {
+        cocaine::logging::log_t& log;
+        zookeeper::connection_t& zk;
+    };
+
+    typedef std::shared_ptr <zookeeper::handler_scope_t> scope_ptr;
+
+    /**
+    * Typedefs for result type. Actual result types are in include/cocaine/idl/unicorn.hpp
+    */
+    typedef api_t::response response;
+
+    zookeeper_api_t(cocaine::logging::log_t& log, zookeeper::connection_t&);
+
+    virtual void
```

Comment:
I personally think saying explicitly that function is virtual is a good habbit even when it overrides base virtual.